lambdas/service/app.py

```diff
@@ -746,71 +748,119 @@ def validate_filters(filters):
+    for `accessions` has invalid properties `{'foo'}`
+
+    >>> validate_filters('{"accessions": {"is": [{"namespace": "baz","foo": "bar"}]}}') # doctest: +NORMALIZE_WHITESPACE
+    Traceback (most recent call last):
+    ...
+    chalice.app.BadRequestError: BadRequestError: The value of the `is` relation in the `filters` parameter entry
+    for `accessions` has invalid properties `{'foo'}`
     """
     try:
         filters = json.loads(filters)
     except Exception:
-        raise BadRequestError('The `filters` parameter is not valid JSON')
+        raise BRE('The `filters` parameter is not valid JSON')
     if type(filters) is not dict:
-        raise BadRequestError('The `filters` parameter must be a dictionary.')
+        raise BRE('The `filters` parameter must be a dictionary')
     for facet, filter_ in filters.items():
         validate_facet(facet)
         try:
             relation, values = one(filter_.items())
         except Exception:
-            raise BadRequestError(f'The `filters` parameter entry for `{facet}` must be a single-item dictionary.')
+            raise BRE(f'The `filters` parameter entry for `{facet}` '
+                      f'must be a single-item dictionary')
         else:
             if facet == app.service_config.source_id_facet:
                 valid_relations = ('is',)
             else:
                 valid_relations = ('is', 'contains', 'within', 'intersects')
             if relation in valid_relations:
                 if not isinstance(values, list):
-                    raise BadRequestError(
-                        msg=f'The value of the `{relation}` relation in the `filters` parameter '
-                            f'entry for `{facet}` is not a list.')
+                    raise BRE(f'The value of the `{relation}` relation in the `filters` '
+                              f'parameter entry for `{facet}` is not a list')
             else:
-                raise BadRequestError(f'The relation in the `filters` parameter entry for `{facet}`'
-                                      f' must be one of {valid_relations}')
+                raise BRE(f'The relation in the `filters` parameter entry '
+                          f'for `{facet}` must be one of {valid_relations}')
             if relation == 'is':
                 value_types = reify(Union[JSON, PrimitiveJSON])
                 if not all(isinstance(value, value_types) for value in values):
-                    raise BadRequestError('The value of the `is` relation in the `filters` '
-                                          'parameter is invalid.')
+                    raise BRE(f'The value of the `is` relation in the `filters` '
+                              f'parameter entry for `{facet}` is invalid')
             if facet == 'organismAge':
                 validate_organism_age_filter(values)
+            field_type = app.repository_controller.field_type_by_filterable_facet(app.catalog)[facet]
```

Comment:
L and for clarity, despite the caching, this call is not in the right scope. See my commit.

Reply:
We should be ore consistent what we call a facet and what we call a field. We're currently all over the place. A facet is a field that's returned in the termFacets part of the service response.